courses/management/commands/import_courserun.py

```diff
@@ -0,0 +1,215 @@
+    """
+    Creates a courseware object.
+    """
+
+    help = "Creates a course run using details from edX."
+
+    def add_arguments(self, parser) -> None:
+        parser.add_argument(
+            "--courserun",
+            type=str,
+            nargs="?",
+            help="The course run to create. Do not specify with --program.",
+        )
+
+        parser.add_argument(
+            "--program",
+            type=str,
+            nargs="?",
+            help="The program to iterate through. Can be the readable ID or the database ID. Requires --run-tag. Do not specify with --courserun.",
+        )
+
+        parser.add_argument(
+            "--run-tag",
+            type=str,
+            nargs="?",
+            help="The run tag (3T2022, etc.) to use. The + will be appended to this when combining it with the course's readable ID. Requires --program. Do not specify with --courserun.",
+        )
+
+        parser.add_argument(
+            "--live",
+            action="store_true",
+            help="Make the new course run live (defaults to not).",
+        )
+
+        parser.add_argument(
+            "--create-cms-page",
+            action="store_true",
+            help="Create a draft CMS page for the course too",
+        )
+
+        parser.add_argument(
+            "--price",
+            help="Create a matching product with the specified price for the generated course run(s).",
+            type=str,
+            nargs="?",
+        )
+
+    def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
+        edx_course_detail = get_edx_api_course_detail_client()
+        edx_courses = []
+
+        if kwargs["price"] and kwargs["price"].isnumeric():
+            content_type = ContentType.objects.filter(
+                app_label="courses", model="courserun"
+            ).get()
+
+        if kwargs["courserun"] is not None:
+            try:
+                course = edx_course_detail.get_detail(
+                    course_id=kwargs["courserun"],
+                    username=settings.OPENEDX_SERVICE_WORKER_USERNAME,
+                )
+
+                if course is not None:
+                    edx_courses.append(course)
+            except Exception as e:
+                self.stdout.write(
+                    self.style.ERROR(
+                        f"Could not retrieve data for {kwargs['courserun']}: {e}"
+                    )
+                )
+                return False
+        elif kwargs["program"] is not None and kwargs["run_tag"] is not None:
+            if kwargs["program"].isnumeric():
+                program = Program.objects.filter(pk=kwargs["program"]).all()
+            else:
+                program = Program.objects.filter(readable_id=kwargs["program"]).all()
+
+            if len(program) > 1:
+                self.stdout.write(
+                    self.style.ERROR(
+                        f"Program ID {kwargs['program']} is ambiguous - {len(program)} results found."
```

Comment:
How could we get multiple rows when filtering on program ID or readable_id since these are unique? Not sure if I am missing something

Reply:
Both of these are related; I was using a Q object to pull the program info. This has been totally refactored in https://github.com/mitodl/mitxonline/pull/1256/commits/e4f82bfe2db26e18d1510a30e6b05f14e4e0055b .